responsibleai/responsibleai/managers/explainer_manager.py

```diff
@@ -179,9 +179,54 @@ def get_data(self):
         :rtype: List[ModelExplanationData]
         """
         return [
-            self._get_interpret(i) for i in self.get()]
+            self._get_interpret(
+                i, self._evaluation_examples) for i in self.get()]
+
+    def request_explanations(self, local: bool, data: Any):
+        """Return the explanations for a given point(s) .
+
+        :param local: True if local explanations are quested
+                      and False otherwise.
+        :type local: bool
+        :param data: The data point(s) for which the explanations
+                     need to be generated.
+        :type data: Any
+        :return: The serialized version explanations for the
+                 given data point(s).
+        :rtype: Any
+        """
+        if not isinstance(data, pd.DataFrame):
+            raise UserConfigValidationException(
+                'Data is of type {0} but it must be '
+                'a pandas DataFrame.'.format(type(data)))
+
+        if local and data.shape[0] > 1:
+            raise UserConfigValidationException(
+                'Only one row of data is allowed for '
+                'local explanation generation.')
+
+        if self._classes is not None:
+            model_task = ModelTask.Classification
+        else:
+            model_task = ModelTask.Regression
+
+        explainer = MimicExplainer(
```

Comment:
nit: Can this be refactored with the mimic explainer created in the compute method into one common method?  It looks like the code in both places is the same.  Similarly the explain_global method, and probably the new explain_local addition, can be refactored out into the common method.

Reply:
Done.